uiServices/src/services.go

```diff
@@ -98,53 +125,89 @@ func validateJsonWithSchema(config []byte) bool {
 
 func getConfigs(rw http.ResponseWriter, req *http.Request){
 
-    configPathDir := req.Header.Get("configPathDir")
+    configPathDir := getConfigHeader(req)
     configName := chi.URLParam(req, "configName")
 
-    if !strings.HasSuffix(configPathDir, "/"){
-        configPathDir = configPathDir + "/"
-    }
+    validateFileNameAndHeader(rw, req,configPathDir,configName)
 
-    if len(configPathDir) <= 1{
+        file, err := os.Open(fmt.Sprintf("%s%s.xml", configPathDir, configName))
+        if err != nil {
+            logrus.Error("Configuration Name Not Found: " + configPathDir + configName)
+            rw.WriteHeader(http.StatusNotFound)
+            return
+        }
+
+        defer file.Close()
+
+        var config perfTestUtils.Config
+
+        byteValue, err := ioutil.ReadAll(file)
+        if err != nil {
+            rw.WriteHeader(http.StatusInternalServerError)
+            logrus.Error("Cannot Read File")
+            return
+        }
+
+        err = xml.Unmarshal(byteValue, &config)
+        if err != nil {
+            rw.WriteHeader(http.StatusInternalServerError)
+            logrus.Error("Cannot Unmarshall")
+            return
+        }
+
+        configJson, err := json.MarshalIndent(config, "", "")
+        if err != nil {
+            rw.WriteHeader(http.StatusInternalServerError)
+            logrus.Error("Cannot Marshall")
+            return
+        }
+
+        rw.WriteHeader(http.StatusOK)
+        rw.Write(configJson)
+        logrus.Println(string(configJson))
+
+
+}
+
+func putConfigs(rw http.ResponseWriter, req *http.Request) {
+    path := getConfigHeader(req)
+    configName := chi.URLParam(req, "configName")
+
+    validateFileNameAndHeader(rw, req,path,configName)
```

Comment:
don't know if this logic makes sense. if file name or header are invalid, we want to immediately return. I think at the moment, the flow will continue on through the code. perhaps in this `validateFileNameAndHeader` set the response status code as you're doing, then return a boolean and if false, just return.